src/main/java/no/nav/veilarbvedtaksstotte/kafka/KafkaHelsesjekk.java

```diff
@@ -1,43 +1,57 @@
 package no.nav.veilarbvedtaksstotte.kafka;
 
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.binder.MeterBinder;
 import lombok.extern.slf4j.Slf4j;
 import no.nav.common.health.HealthCheck;
 import no.nav.common.health.HealthCheckResult;
-import org.apache.kafka.clients.consumer.Consumer;
-import org.apache.kafka.clients.consumer.ConsumerRecord;
-import org.springframework.kafka.listener.ContainerAwareErrorHandler;
-import org.springframework.kafka.listener.MessageListenerContainer;
+import no.nav.veilarbvedtaksstotte.repository.KafkaRepository;
+import no.nav.veilarbvedtaksstotte.repository.domain.MeldingType;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
-import java.util.Date;
-import java.util.List;
+import static java.lang.String.format;
 
 @Slf4j
 @Component
-public class KafkaHelsesjekk implements HealthCheck, ContainerAwareErrorHandler {
+public class KafkaHelsesjekk implements MeterBinder, HealthCheck {
 
-    private static final long EN_MINUTT = 60_000L;
+    private final KafkaRepository kafkaRepository;
 
-    private volatile long lastThrownExceptionTime;
-
-    private volatile Exception lastThrownException;
+    @Autowired
+    public KafkaHelsesjekk(KafkaRepository kafkaRepository) {
+        this.kafkaRepository = kafkaRepository;
+    }
 
     @Override
     public HealthCheckResult checkHealth() {
-        boolean hasFailedRecently = (lastThrownExceptionTime + EN_MINUTT) > System.currentTimeMillis();
+        int totaltFeiledeMeldinger = hentFeiledeKafkaMeldinger(MeldingType.CONSUMED)
+                + hentFeiledeKafkaMeldinger(MeldingType.PRODUCED);
 
-        if (hasFailedRecently) {
-            return HealthCheckResult.unhealthy("Kafka consumer feilet " + new Date(lastThrownExceptionTime), lastThrownException);
+        if (totaltFeiledeMeldinger > 0) {
+            return HealthCheckResult.unhealthy(format("Det ligger lagret %d kafka meldinger som ikke er konsumert/publisert", totaltFeiledeMeldinger));
```

Comment:
Nice to have; se fordeling på feiler for produsent og konsument? Kanskje per topic også?